src/analog/dac.rs

```diff
@@ -14,16 +14,40 @@ pub trait DacOut<V> {
     fn get_value(&mut self) -> V;
 }
 
+pub struct GeneratorMode {
+    mode: u8,
+    amp: u8,
+}
+
+impl GeneratorMode {
+    pub fn triangle(amplitude: u8) -> Self {
+        Self {
+            mode: 0b10,
```

Comment:
What do you think, is it better to use some `enum` here to make it more explicit what it can be?

Reply:
@andresv it was my first idea to use enum, but generator has 2 modes(Noise & Triangles) and for each mode we need to provide additional data(amplitude for triangles and seed for noise). 

Of course we can use something like that:
```rust
enum GeneratorMode {
  Noise(u8),
  Triangles(u8)
}
```
but meaning of enum parameters here is unclear.

Anyway, I plan to add event support to control ADC from timers, and I think current GeneratorMode gonna be a good place for events setup as well.

What do you think about this?

Reply:
Got it.

Looking forward to see proper ADC control from timers. Currently I am using a hack to trigger ADC sequencing over DMA using a timer input.